### PR TITLE
BUG: np.nonzero behaving differently for 1-D arrays

### LINKED ISSUE #5837: numpy.where's result is read-only for multi-dimensional arrays
See http://stackoverflow.com/questions/28663142/why-is-np-wheres-result-read-only-for-multi-dimensional-arrays

### PR DESCRIPTION
**Pull request by jaimefrio**
Fixes #5837

### CODE REVIEW

**Comment by jaimefrio** [comment]
Some sample output after and before this PR to ilustrate the differences:

```
>>> class C(np.ndarray): pass
... 
>>> a = np.arange(6).view(C)
>>> b = np.arange(6).reshape(2, 3).view(C)
>>> anz = a.nonzero()
>>> bnz = b.nonzero()

###    THIS PR    ###      ###     MASTER     ###
>>> type(anz[0])
<type 'numpy.ndarray'>     <type 'numpy.ndarray'>
>>> anz[0].flags
  C_CONTIGUOUS : True        C_CONTIGUOUS : True     
  F_CONTIGUOUS : True        F_CONTIGUOUS : True
  OWNDATA : False            OWNDATA : True
  WRITEABLE : True           WRITEABLE : True
  ALIGNED : True             ALIGNED : True
  UPDATEIFCOPY : False       UPDATEIFCOPY : False
>>> anz[0].base
array([[1],                None
       [2],
       [3],
       [4],
       [5]])
###    THIS PR    ###      ###     MASTER     ###
>>> type(bnz[0])
<type 'numpy.ndarray'>     <class '__main__.C'>
>>> bnz[0].flags
  C_CONTIGUOUS : False       C_CONTIGUOUS : False
  F_CONTIGUOUS : False       F_CONTIGUOUS : False
  OWNDATA : False            OWNDATA : False
  WRITEABLE : True           WRITEABLE : False
  ALIGNED : True             ALIGNED : True
  UPDATEIFCOPY : False       UPDATEIFCOPY : False
>>> bnz[0].base
array([[0, 1],             array([[0, 1],
       [0, 2],                    [0, 2],
       [1, 0],                    [1, 0],
       [1, 1],                    [1, 1],
       [1, 2]])                   [1, 2]])

```

File: doc/release/1.10.0-notes.rst [hunk 1]
```diff
@@ -86,6 +97,8 @@ provided in the 'out' keyword argument, and it would be used as the first
 output for ufuncs with multiple outputs, is deprecated, and will result in a
 `DeprecationWarning` now and an error in the future.
 
+of indices into the input, it was decided to return
+
```
Comment (njsmith) [inline doc/release/1.10.0-notes.rst]: ?

File: numpy/core/tests/test_numeric.py [hunk 2]
```diff
@@ -951,6 +951,20 @@ def test_sparse(self):
...
+
+        for view in (C, np.ndarray):
+            for nd in range(1, 4):
+                shape = tuple(range(2, 2+nd))
+                x = np.arange(np.prod(shape)).reshape(shape).view(view)
+                for nzx in (np.nonzero(x), x.nonzero()):
+                    for nzx_i in nzx:
+                        assert_(type(nzx_i) is np.ndarray)
+                        assert_(nzx_i.flags.writeable)
+                        assert_(nzx_i.base is not None)
```
Comment (njsmith) [inline numpy/core/tests/test_numeric.py]: Is there any need for the view check? I don't think this is part of the public API, i.e. if we changed it so it was no longer a view, that would not be a bug, so the tests probably should not flag it as one.

**Comment by njsmith** [comment]
Otherwise LGTM

**Comment by jaimefrio** [comment]
Addressed comments, ready for review.

**Comment by njsmith** [comment]
LGTM, but I'll let travis check me on that

**Comment by charris** [comment]
@njsmith Finish this up please.

**Comment by charris** [comment]
Thanks Jaime.

**Pull request merged**